admin_guide/persistent_storage_nfs.adoc

```diff
@@ -74,30 +56,86 @@ persistent-volume.json
 ----
 ====
 
+=== Enforcing Disk Quotas
+Use disk partitions to enforce disk quotas and size constraints. Each partition
+can be its own export. Each export is one persistent volume. Kubernetes enforces
+unique names for persistent volumes, but the uniqueness of the NFS volume's
+server and path is up to the administrator.
+
+Enforcing quotas in this way allows the end user to request persistent storage
+by a specific amount (e.g,, 10Gi) and be matched with a corresponding volume of
+equal or greater capacity.
+
+=== Volume Security
+Users request storage with a `*PersistentVolumeClaim*`. This claim only lives in
+the user's namespace and can only be referenced by a pod within that same
+namespace. Any attempt to access a persistent volume across a namespace causes
+the pod to fail.
+
+Each NFS volume must be mountable by all nodes in the cluster.
+
+== Reclaiming Resources
+NFS implements the Kubernetes *Recyclable* plug-in interface. Automatic
+processes handle reclamation tasks based on policies set on each persistent
+volume.
+
+By default, persistent volumes are set to *Retain*. NFS volumes which are set to
+*Recycle* are scrubbed (i.e., `rm -rf` is run on the volume) after being
+released from their claim (i.e, after the user's `*PersistentVolumeClaim*` bound
+to the volume is deleted). Once recycled, the NFS volume can be bound to a new
+claim.
 
 == Automation
+As discussed, clusters can be provisioned with persistent storage using NFS in
+the following way:
+
+- Disk partitions can be used to link:#enforcing-disk-quotas[enforce storage
+quotas].
+- Security can be enforced by link:#volume-security[restricting volumes] to the
+namespace that has a claim to them.
+- link:#reclaiming-resources[Reclamation of discarded resources] can be
+configured for each persistent volume.
 
-In this topic, we covered how a cluster can be provisioned with persistent storage using NFS.  Disk partitions can be
-used to enforce storage quotas, security is enforced by restricting volumes to the namespace that has a claim to them, and
-reclamation of discarded resources can be configured for each persistent volume.
+They are many ways that you can use scripts to automate the above tasks. You can
+use an
+link:https://github.com/openshift/openshift-ansible/tree/master/roles/kube_nfs_volumes[example
+Ansible playbook] to help you get started.
+
+== SELinux and NFS Export Settings
+By default, SELinux does not allow writing from a pod to a remote NFS server.
+The NFS volume mounts correctly, but is read-only.
+
+To enable writing in SELinux on each node:
+
+----
+# setsebool -P virt_use_nfs 1 <1>
+----
+<1> `-P` makes the bool persistent between reboots.
 
-Many ways to script these tasks exist.  We've created an link:https://github.com/openshift/openshift-ansible/tree/master/roles/kube_nfs_volumes[Ansible playbook] to help you get started.
+Additionally, each exported volume on the NFS server itself should conform to
+the following:
```

Comment:
I think a little justification would be helpful here. The problem all this addresses is that we have no idea which user the containers will end up running as, so the administrator doesn't know a priori how to set the permissions on the volume. The problem isn't evident to administrators until apps start trying to use the volume and fail, and their owners complain.

So I think it would be useful to insert:

    Additionally, in order to enable arbitrary container users to read and write the volume,
    each exported volume on the NFS server itself should conform to the following: